packages/core/lib/configureGanacheOptions.js

```diff
@@ -18,27 +18,27 @@ function sanitizeGanacheOptions(ganacheOptions) {
 
 function configureManagedGanache(config, networkConfig, mnemonic) {
   const calcTotalAccounts = networkConfig => {
-    // Respect user's number of accounts choice to 0 in truffle config
-    const isZeroAccount =
-      networkConfig.accounts === 0 || networkConfig.total_accounts === 0;
-    const userAccounts = networkConfig.accounts || networkConfig.total_accounts;
-    return isZeroAccount ? 0 : userAccounts || 10;
+    if ("accounts" in networkConfig) {
+      return networkConfig.accounts;
+    }
+    if ("total_accounts" in networkConfig) {
+      return networkConfig.total_accounts;
+    }
+    return 10; // Default number of accounts when undefined in the truffle config
   };
```

Comment:
@haltman-at , I found this way a bit cleaner to understand the intended behavior. Let me know if this makes sense to you.